src/edu/csus/ecs/pc2/ui/ShadowControlPane.java

```diff
@@ -713,8 +804,57 @@ public void run() {
                                     boolean isConnected = remoteContestAPIAdapter.testConnection();
                                     if (isConnected) {
                                         showMessage ("Connection to remote CCS is successful");
+                                        addConnectTableEntry("Test connection to remote CCS successful");
                                     } else {
                                         showErrorMessage("Connection to remote CCS failed", "Connection failed");
+                                        addConnectTableEntry("Test connection to remote CCS failed");
+                                    }
+                                    
+                                    // API endpoint is right before /contests/ in the URL, so find that, if it's there
```

Comment:
Does this directly involve the connection logging?
The block staring  // API endpoint is right before /contests/ in the URL, so find that, if it's there
This block is too long for the method, it should be put into its own method, also this likely will
provide a way to unit test this block.

Reply:
This change has been made and the code has been moved into separate methods.